src/main/java/tc/oc/pgm/match/MatchPlayerImpl.java

```diff
@@ -172,7 +174,13 @@ public boolean canInteract() {
 
   @Override
   public boolean canSee(MatchPlayer other) {
-    return (isObserving() || other.isParticipating()) && other.isVisible();
+    if (!other.isVisible()) return false;
+    if (other.isParticipating()) return true;
+    if (isObserving()) {
+      SettingValue observers = getSettings().getValue(SettingKey.OBSERVERS);
+      return observers.equals(SettingValue.OBSERVERS_SHOWN);
+    }
+    return false;
```

Comment:
Change to:
```java
return getSettings().getValue(SettingKey.OBSERVERS) == SettingValue.OBSERVERS_SHOWN;
```

Reply:
Actually, if you make it that, players inside the match will be able to see obs.